Src/Base/AMReX_Algorithm.H

```diff
@@ -157,6 +157,58 @@ namespace amrex
         return hi;
     }
 
+    template<typename ItType, typename ValType>
+    AMREX_GPU_DEVICE
+    ItType upper_bound (ItType first, ItType last, const ValType& val)
+    {
+#if AMREX_DEVICE_COMPILE
+        ItType it;
+        size_t count, step;
+        count = last-first;
+        while(count>0){
+            it = first;
+            step = count/2;
+            it += step;
+            if (!(val<*it)){
+                first = ++it;
+                count -= step + 1;
+            }
+            else{
+                count = step;
+            }
+        }
+
+        return first;
+#else
+        return std::upper_bound(first, last, val);
+#endif
+    }
+
+    template<typename ItType, typename ValType>
+    AMREX_GPU_DEVICE
+    ItType lower_bound (ItType first, ItType last, const ValType& val)
+    {
+#ifdef AMREX_DEVICE_COMPILE
+        size_t count = last-first;
+        do{
```

Comment:
The implementation here uses `do-while`, whereas `upper_bound` uses `while`.  Maybe this is correct.  But why not using the same style as `upper_bound` (which seems the style used by gcc and llvm)?